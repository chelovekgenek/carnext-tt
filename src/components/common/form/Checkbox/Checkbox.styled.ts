import styled from "styled-components"

export const Checkbox = styled.input`
  height: 1.2rem;
  width: 1.2rem;
  margin: 0 0.6rem 0 0;
`

export const Label = styled.label`
  font-size: 1.5rem;
`
