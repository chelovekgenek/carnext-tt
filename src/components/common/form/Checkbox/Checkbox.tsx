import React from "react"
import { Field, FieldProps } from "formik"

import { FormItem, FormItemError } from "../FormItem"

import * as Styles from "./Checkbox.styled"

interface IProps {
  name: string
  label: string
}

export const Checkbox: React.FC<IProps> = ({ name, label }) => (
  <Field name={name}>
    {(formikFieldProps: FieldProps) => <InnerFormItem name={name} label={label} {...formikFieldProps} />}
  </Field>
)

export const InnerFormItem: React.FC<IProps & FieldProps> = ({
  label,
  name,
  field,
  form: { touched, errors },
  ...props
}) => (
  <FormItem>
    <Styles.Checkbox id={name} type="checkbox" checked={field.value || false} {...field} {...props} />
    <Styles.Label htmlFor={name}>{label}</Styles.Label>
    <FormItemError>{touched[name] && errors[name]}</FormItemError>
  </FormItem>
)
