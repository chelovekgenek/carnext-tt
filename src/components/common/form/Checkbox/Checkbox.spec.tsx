import React from "react"
import { shallow } from "enzyme"

import fieldProps from "mocks/formik-field-props"

import { InnerFormItem } from "./Checkbox"

describe("Checkbox", () => {
  const defaultProps = {
    ...fieldProps,
    field: { ...fieldProps.field, value: false, name: "checkbox" },
    name: "checkbox",
    label: "checkbox",
  }
  const render = (props: React.ComponentProps<typeof InnerFormItem> = defaultProps) =>
    shallow(<InnerFormItem {...props} />)

  it("should render", () => {
    expect(render()).toMatchSnapshot()
  })
})
