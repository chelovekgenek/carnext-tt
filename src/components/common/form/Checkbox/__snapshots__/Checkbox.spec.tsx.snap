// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render 1`] = `
<styled.div>
  <styled.input
    checked={false}
    id="checkbox"
    name="checkbox"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    type="checkbox"
    value={false}
  />
  <styled.label
    htmlFor="checkbox"
  >
    checkbox
  </styled.label>
  <styled.div />
</styled.div>
`;
